File: ios/Classes/Methods.swift
//
// Created by Bog'ibek on 22/10/25.
//

import Foundation

enum Methods {
    // Platform
    static let INITIALIZE = "initialize"
    static let GET_PLATFORM_VERSION = "getPlatformVersion"
    static let INIT_OPTIONS         = "init_options"
    static let SAVE_LANGUAGE        = "save_language"
    static let CHANGE_SOUND_ENABLE  = "change_sound_enable"

    // Channels
    static let CREATE_CHANNEL       = "create_channel"
    static let DELETE_CHANNEL       = "delete_channel"
    static let CHANNEL_EXISTS       = "channel_exists"
    static let GET_ALL_CHANNELS     = "get_all_channels"

    // Notifications
    static let SHOW_NOTIFICATION    = "show_notification"
    static let CANCEL_NOTIFICATION  = "cancel_notification"
    static let CANCEL_ALL           = "cancel_all_notifications"
}

File: android/src/main/kotlin/org/perfect/notifications/perfect_notifications/enum/Methods.kt
package org.perfect.notifications.perfect_notifications.enum

object Methods {
    const val INITIALIZE   = "initialize"
    const val GET_PLATFORM_VERSION   = "getPlatformVersion"
    const val INIT_OPTIONS           = "init_options"
    const val SAVE_LANGUAGE           = "save_language"

    const val CHANGE_SOUND_ENABLE     = "change_sound_enable"

    // Channels
    const val CREATE_CHANNEL         = "create_channel"
    const val DELETE_CHANNEL         = "delete_channel"
    const val CHANNEL_EXISTS         = "channel_exists"
    const val GET_ALL_CHANNELS       = "get_all_channels"

    // Notifications
    const val SHOW_NOTIFICATION      = "show_notification"
    const val CANCEL_NOTIFICATION    = "cancel_notification"
    const val CANCEL_ALL             = "cancel_all_notifications"

}